launcher/ui/pages/modplatform/ImportPage.cpp

```diff
@@ -116,28 +121,75 @@ void ImportPage::updateState()
             // mrpack is a modrinth pack
             bool isMRPack = fi.suffix() == "mrpack";
 
-            if(fi.exists() && (isZip || isMRPack))
-            {
+            if (fi.exists() && (isZip || isMRPack)) {
                 QFileInfo fi(url.fileName());
-                dialog->setSuggestedPack(fi.completeBaseName(), new InstanceImportTask(url,this));
+                auto extra_info = QMap(m_extra_info);
+                qDebug() << "Pack Extra Info" << extra_info << m_extra_info;
+                dialog->setSuggestedPack(fi.completeBaseName(), new InstanceImportTask(url, this, std::move(extra_info)));
                 dialog->setSuggestedIcon("default");
             }
-        }
-        else
-        {
-            if(input.endsWith("?client=y")) {
+        } else if (url.scheme() == "curseforge") {
```

Comment:
i am confused, this code seems very similar to the one in `MainWindow.cpp`. Can we make them share this code in some way?

Reply:
as similar as they appear the two contain subtly different "succeed" blocks because one is generic to any given curseforge resource and the other is for modpacks only. The shared code is effectively control flow and as nice as DRY is trying to share code here would be a premature and unneeded abstraction IMO.